internal/clusterfeature/features/vault/client.go

```diff
@@ -0,0 +1,108 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vault
+
+import (
+	"fmt"
+	"os"
+
+	"emperror.dev/errors"
+	"github.com/banzaicloud/bank-vaults/pkg/sdk/vault"
+	vaultapi "github.com/hashicorp/vault/api"
+)
+
+type vaultManager struct {
+	vaultClient *vault.Client
+}
+
+func newVaultManager(spec vaultFeatureSpec, orgID, clusterID uint) (*vaultManager, error) {
+	var vaultAddress string
+	if spec.CustomVault.Enabled {
+		vaultAddress = spec.CustomVault.Address
+	} else {
+		vaultAddress = os.Getenv(vaultAddressEnvKey)
+	}
+
+	clientConfig := vaultapi.DefaultConfig()
+	clientConfig.Address = vaultAddress
+
+	var clientOptions = []vault.ClientOption{
+		vault.ClientRole(roleName),
+		vault.ClientAuthPath(getAuthMethodPath(orgID, clusterID)),
+	}
+	if len(spec.CustomVault.Token) != 0 {
```

Comment:
I think in this case we need to ask for a Vault path as well, where the user would like to set the policy to enable the reading of Secrets (only in case of custom Vault).